### PR TITLE
Add ocne start cluster --provider olvm

### PR DESCRIPTION
**Pull request by pfmackin**
Add ocne start cluster --provider olvm

1. This code is optional enabled only for develop builds
2. This feature depends on the OLVM CAPI provider

Changes:

1. added OLVM CAPI driver
2. added support for OLVM CAPI template building.

Testing:

**Developer build disabled**
```
 ocne cluster start  --kubeconfig=/Users/pmackin/.kube/kubeconfig.ocne.local --provider olvm  --cluster-name paul -c /Users/pmackin/.ocne/olvm-gz-1cp-1w.yaml
ERRO[2024-12-11T10:07:15-05:00] No implementation exists for the olvm driver 
```

**Provision 3 CP and 3 worker nodes**
````
ocne cluster start  --kubeconfig=/Users/pmackin/.kube/kubeconfig.ocne.local --provider olvm  --cluster-name paul -c /Users/pmackin/.ocne/olvm-gz-3cp-1worker.yaml
INFO[2024-12-09T15:29:25-05:00] Installing cert-manager into cert-manager: ok 
INFO[2024-12-09T15:29:26-05:00] Installing core-capi into capi-system: ok 
INFO[2024-12-09T15:29:26-05:00] Installing olvm-capi into cluster-api-provider-olvm: ok 
INFO[2024-12-09T15:29:27-05:00] Installing bootstrap-capi into capi-kubeadm-bootstrap-system: ok 
INFO[2024-12-09T15:29:27-05:00] Installing control-plane-capi into capi-kubeadm-control-plane-system: ok 
INFO[2024-12-09T15:29:28-05:00] Waiting for Core Cluster API Controllers: ok 
INFO[2024-12-09T15:29:28-05:00] Waiting for Kubadm Boostrap Cluster API Controllers: ok 
INFO[2024-12-09T15:29:28-05:00] Waiting for Kubadm Control Plane Cluster API Controllers: ok 
INFO[2024-12-09T15:29:28-05:00] Applying Cluster API resources               
INFO[2024-12-09T15:30:00-05:00] Waiting for kubeconfig: ok       
INFO[2024-12-09T15:32:32-05:00] Waiting for the Kubernetes cluster to be ready: ok 
INFO[2024-12-09T15:32:32-05:00] Installing applications into workload cluster 
INFO[2024-12-09T15:32:37-05:00] Installing flannel into kube-flannel: ok 
INFO[2024-12-09T15:32:39-05:00] Installing ui into ocne-system: ok 
INFO[2024-12-09T15:32:40-05:00] Installing ocne-catalog into ocne-system: ok 
INFO[2024-12-09T15:32:40-05:00] Kubernetes cluster was created successfully  
INFO[2024-12-09T15:36:11-05:00] Waiting for the UI to be ready: ok 


**kubeclt get the nodes**
 k get node -o wide
NAME                       STATUS   ROLES           AGE     VERSION         INTERNAL-IP      EXTERNAL-IP   OS-IMAGE                   KERNEL-VERSION                     CONTAINER-RUNTIME
paul-control-plane-xyz   Ready    control-plane   2m28s   v1.30.3+1.el8   1.2.3.4 (redacted)   <none>        Oracle Linux Server 8.10   5.15.0-209.161.7.2.el8uek.x86_64   cri-o://1.30.3
paul-control-plane-abc   Ready    control-plane   4m59s   v1.30.3+1.el8   1.2.3.5 (redacted)   <none>        Oracle Linux Server 8.10   5.15.0-209.161.7.2.el8uek.x86_64   cri-o://1.30.3
paul-control-plane-123   Ready    control-plane   8m3s    v1.30.3+1.el8   1.2.3.6 (redacted)    <none>        Oracle Linux Server 8.10   5.15.0-209.161.7.2.el8uek.x86_64   cri-o://1.30.3
paul-md-0-79mch-tsx      Ready    <none>          4m1s    v1.30.3+1.el8   1.2.3.7 (redacted)    <none>        Oracle Linux Server 8.10   5.15.0-209.161.7.2.el8uek.x86_64   cri-o://1.30.3
paul-md-0-79mch-tsd      Ready    <none>          5m27s   v1.30.3+1.el8  1.2.3.8 (redacted)    <none>        Oracle Linux Server 8.10   5.15.0-209.161.7.2.el8uek.x86_64   cri-o://1.30.3
paul-md-0-79mch-sfs      Ready    <none>          3m54s   v1.30.3+1.el8  1.2.3.9 (redacted)    <none>        Oracle Linux Server 8.10   5.15.0-209.161.7.2.el8uek.x86_64   cri-o://1.30.3
````

**Delete cluster**
```
 ocne cluster delete  --kubeconfig=/Users/pmackin/.kube/kubeconfig.ocne.local --cluster-name paul
INFO[2024-12-11T17:02:55-05:00] Installing cert-manager into cert-manager: ok 
INFO[2024-12-11T17:02:56-05:00] Installing core-capi into capi-system: ok 
INFO[2024-12-11T17:02:56-05:00] Installing olvm-capi into cluster-api-provider-olvm: ok 
INFO[2024-12-11T17:02:57-05:00] Installing bootstrap-capi into capi-kubeadm-bootstrap-system: ok 
INFO[2024-12-11T17:02:57-05:00] Installing control-plane-capi into capi-kubeadm-control-plane-system: ok 
INFO[2024-12-11T17:02:58-05:00] Waiting for Core Cluster API Controllers: ok 
INFO[2024-12-11T17:02:58-05:00] Waiting for Kubadm Boostrap Cluster API Controllers: ok 
INFO[2024-12-11T17:02:58-05:00] Waiting for Kubadm Control Plane Cluster API Controllers: ok 
INFO[2024-12-11T17:02:58-05:00] Deleting Cluster olvm-cluster/paul           
INFO[2024-12-11T17:03:19-05:00] Waiting for deletion: ok  
```

**kubeclt get the nodes**
```
 k get node -o wide -w
E1211 17:04:25.186461   88497 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp [::1]:8080: connect: connection refused
```

**ocne template command**
```
 ocne cluster template  --kubeconfig=/Users/pmackin/.kube/kubeconfig.ocne.local --provider olvm   -c /Users/pmackin/.ocne/olvm-gz-3cp-3w.yaml
 
 apiVersion: cluster.x-k8s.io/v1beta1
kind: Cluster
metadata:
  labels:
    cluster.x-k8s.io/cluster-name: "paul"
  name: "paul"
  namespace: "olvm-cluster"
spec:
  clusterNetwork:
    pods:
      cidrBlocks:
        - 10.244.0.0/16
    serviceDomain: "cluster.local"
    services:
      cidrBlocks:
        - 10.96.0.0/12
  infrastructureRef:
    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
    kind: OLVMCluster
    name: "paul"
    namespace: "olvm-cluster"
  controlPlaneRef:
    apiVersion: controlplane.cluster.x-k8s.io/v1beta1
    kind: KubeadmControlPlane
    name: "paul-control-plane"
    namespace: "olvm-cluster"
---
apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
kind: OLVMCluster
metadata:
  labels:
    cluster.x-k8s.io/cluster-name: "paul"
  name: "paul"
  namespace: "olvm-cluster"
spec:
  ovirtDatacenterName: "Default"
  controlPlaneEndpoint:
    host: "1.2.3.4"
    port: 6443
  ovirtAPI:
    apiServerURL: "https://xyz"
    caConfigMapName: "paul-ovirt-ca"
    credentialsSecretName: "paul-ovirt-credentials"
  vmIpProfiles:
    - name: "default-ip"
      startingIpAddress: "1.3.4.5"
      gateway: "1.2.3.4"
      netmask: "255.255.255.0"
      device: "enp1s0"
---
apiVersion: controlplane.cluster.x-k8s.io/v1beta1
kind: KubeadmControlPlane
metadata:
  name: "paul-control-plane"
  namespace: "olvm-cluster"
spec:
  version: "1.30.3"

...
```

### CODE REVIEW

File: pkg/cluster/driver/olvm/apps.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	proxyValues := map[string]interface{}{
+		"httpsProxy": cad.ClusterConfig.Providers.Olvm.Proxy.HttpsProxy,
+		"httpProxy":  cad.ClusterConfig.Providers.Oci.Proxy.HttpProxy,
+		"noProxy":    cad.ClusterConfig.Providers.Oci.Proxy.NoProxy,
+	}
+
+	return []install.ApplicationDescription{
+		install.ApplicationDescription{
+			Application: &types.Application{
+				Name:      constants.CertManagerChart,
```
Comment (mgianatagh) [inline pkg/cluster/driver/olvm/apps.go:27]: A similar set of apps is installed for oci cluster api.  Could there be a common set the code shares?

Comment (pfmackin) [reply pkg/cluster/driver/olvm/apps.go:27]: Yes there is lots of common code between OCI capi and OLVM capi.  The plan is to refactor when we do the next CAPI provider and have common code that all the providers share.

File: pkg/cluster/driver/olvm/driver.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+func (cad *OlvmDriver) Stop() error {
+	return fmt.Errorf("OlvmDriver.Stop() is not implemented")
+}
+
+func (cad *OlvmDriver) GetKubeconfigPath() string {
+	return cad.KubeConfig
+}
+
+func (cad *OlvmDriver) GetKubeAPIServerAddress() string {
```
Comment (dkrasins) [inline pkg/cluster/driver/olvm/driver.go:230]: The kube-proxy stuff I'm doing now relies on this value.  It should point to the IP of the Kubernetes API Server (virtualIp or loadBalancer).

File: pkg/constants/k8s_constants.go [hunk 3]
```diff
@@ -96,6 +96,20 @@ const (
...
+	OLVMCAPIOperatorNamespace  = "cluster-api-provider-olvm"
+	OLVMCAPIChart              = "olvm-capi"
+	OLVMCAPIVersion            = ""
+	OLVMCAPIDeployment         = "olvm-capi-operator"
+	OLVMOVirtCredSecretSuffix  = "ovirt-credentials"
+	OLVMOVirtCAConfigMapSuffix = "ovirt-ca"
+
+	// OLVM CAPI resources constants
+	OLVMCAPIResourcesNamespace = "ocne"
+	OLVMCAPIControlPlaneMemory = "7GB"
```
Comment (dkrasins) [inline pkg/constants/k8s_constants.go:110]: Did you mean 8?

Comment (pfmackin) [reply pkg/constants/k8s_constants.go:110]: no i have been testing with 7

File: pkg/cluster/driver/olvm/driver.go (hunk 2 above)
Comment (pfmackin) [reply pkg/cluster/driver/olvm/driver.go:231]: fixed, returns http://<ip>:<port>

**Review by mgianatagh — APPROVED** [review]

**Review by dkrasins — APPROVED** [review]

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**